library/tiqr/Tiqr/Random.php

```diff
@@ -21,51 +21,31 @@
 
 /**
  * A class implementing secure random number generation.
- * If openssl functionality is available, openssl is used to generate 
- * secure random data, if not, a twisted sha1 hash of a random number is used.
- * 
+ *
  * @author ivo
  *
  */
 class Tiqr_Random
 {
     /**
-     * Generate $length random bytes.
-     * 
-     * Code courtesy of http://www.zimuel.it/blog/2011/01/strong-cryptography-in-php/
-     * 
+     * Generate $length cryptographically secure pseudo-random bytes
+     * Throws when requested number of bytes cannot be generated
+     *
      * @param int $length the number of bytes to generate.
+     * @return string containing $length cryptographically secure pseudo-random bytes
+     *
+     * @throws Exception, Error, TypeError
      */
     public static function randomBytes($length)
     {
-       if(function_exists('openssl_random_pseudo_bytes')) {
-            $rnd = openssl_random_pseudo_bytes($length, $strong);
-            if($strong === TRUE && $rnd !== FALSE) {
-                return $rnd;
-            }
-        }
+        // Get $length cryptographically secure pseudo-random bytes
+        $rnd=\random_bytes($length);
 
-        // When openssl_random_pseudo_bytes failed, fall back on a mt_rand based string.
-
-        $rnd='';
-        
-        for ($i=0;$i<$length;$i++) {
-            $sha= sha1(mt_rand());
-            $char= mt_rand(0,30);
-            $rnd.= chr(hexdec($sha[$char].$sha[$char+1]));
+        if (strlen($rnd) != $length) {
```

Comment:
Please opt for using the (not) identical equality operator: `!==`. This to prevent false positives due to PHP's duck typing.